Software/Source/macidarray.py

```diff
@@ -0,0 +1,36 @@
+import json
+macid=[]
+
+def searchmacid():
+	f=open("Non-Connected_Data_G2.txt","r")
+	while True:	
+		content=f.readline()	
+		# print(content)
+		if content:
+			file_obj = json.loads(content)
+			file_mac = file_obj["SensorUUID"]
+		
+		else:
+			break	
+		
+		find = checkmacid(file_mac)
+		if find == 0:
+			macid.append(file_mac)
+			print(macid)
+		
+			
+	return macid
+
+#To check macid
+def checkmacid(file_mac):
+	n = 0
+	global macid
+	for i in range (len(macid)):
+		if macid[i] == file_mac:
+			n = i
+	return n

```

Comment:
there is logical error here, you are doing return n and in while loop checking for "find == 0"

Reply:
@ManjulaMN  : Execute the code locally and check it is functionally correct. As suggested by @prashantachari , i also see errors. Check and then upload